### PR TITLE
Fix #206: migrate to bintray

### LINKED ISSUE #206: Migrate from Sonatype to Bintray to publish the library in maven central
Migrate from Sonatype to Bintray to publish the library in maven central

### PR DESCRIPTION
**Pull request by maxme**
Fix #206: migrate to bintray

Notes:
- I had to use bintray plugin version 0.4.0, more recent versions requires to upgrade gradle + gradle android plugin.
- I ignored Javadoc warnings in 0cb56f0 but we might want to fix them (the doc is bundled and uploaded to the maven repo).
- I set up the default publication to `clientRelease` in 65e1787.
- `./gradlew connectedCheck` fails in `develop` and in this branch, I haven't tried to fix them.

### CODE REVIEW

File: Simperium/build.gradle [hunk 1]
```diff
@@ -97,69 +94,16 @@ public final class Version {
...
-            }
-        }
-    }
+publish {
+    artifactId = 'simperium'
+    userOrg = 'simperium'
+    groupId = 'com.simperium.android'
+    uploadName = 'simperium'
+    description = 'Android Library for building applications using Simperium'
+    publishVersion = android.defaultConfig.versionName
```
Comment (maxme) [inline Simperium/build.gradle]: Looking at previous version names here https://repo1.maven.org/maven2/com/simperium/android/simperium/ - we should probably change how we set `publishVersion` here, maybe having a simple string "0.7.2" for now?

Comment (theck13) [reply Simperium/build.gradle]: Using a string like `'0.7.2'` works for me.  Would `publishVersion = project.version` work too?  Either way is fine with me.

**Review by theck13 — COMMENTED** [review]
Everything looks good to me.  The build ran successfully with the `./gradlew assemble publishToMavenLocal bintrayUpload -PbintrayUser=CHANGEME -PbintrayKey=CHANGEME -PdryRun=true` command.

I left one comment in reply to your comment.  Let me know which way you want to handle the version name.

File: Simperium/build.gradle (hunk 1 above)
Comment (maxme) [reply Simperium/build.gradle]: `project.version` and `android.defaultConfig.versionName` is the same string, and comes from the `git describe` command which return the latest tag + number of changes + a hash. We could get the latest tag only and even check the current commit was tagged and fail if not.

I prefer having a simple string here as it's more flexible and we won't release it often. Updated in 46d2e92

**Review by theck13 — APPROVED** [review]

**Pull request merged**